docs/tasks/stateful-sets/deleting-pods.md

```diff
@@ -0,0 +1,70 @@
+---
+assignees:
+- bprashanth
+- erictune
+- foxish
+- smarterclayton
+
+---
+
+{% capture overview %}
+This page shows how to delete pods which are part of a stateful set, and explains the various considerations when doing so.
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* If you are not using Stateful Sets, you can skip this task now. 
+* This is a fairly advanced task and has the potential to violate some properties provided by Stateful Sets.
+* If you are attempting this, you must be well aware of the various considerations enumerated below, and have specific domain knowledge of the application(s) running using Stateful Set.
+
+{% endcapture %}
+
+{% capture steps %}
+
+
+### Stateful Set considerations
+
+In normal operation of a Stateful Set, there is **never** a need to force delete a Stateful Set pod. The Stateful Set controller is responsible for creating, scaling and deleting members of the Stateful Set. It tries to ensure that the specified number of Pods from ordinal 0 through N-1 are alive and ready. Stateful Set also gives us the property that there is at any time at most one pod with a given identity running in a cluster. This is  referred to as *at most one* semantics provided by a Stateful Set. (*TODO: Link to concept page.*)
+
+Manual force deletion is a maintenance task that must be undertaken with caution. It has the potential to violate *at most one* semantics provided by Stateful Sets. Stateful Sets may be used to run distributed and clustered applications which have a need for a stable network identity and stable storage. These applications often have configuration which relies on an ensemble of a fixed number of members with fixed identities. Having multiple members with the same identity can be disastrous and may lead to data loss (e.g. split brain scenario in quorum-based systems).  
+
+### Deletion of Pods
+
+#### Graceful Deletion
+
+Graceful deletion may be performed using the following:
+
+```shell
+kubectl delete pods <pod>
+```
+
+For the above to lead to graceful termination, the pod **must not** specify a `pod.Spec.TerminationGracePeriodSeconds` of 0. The practice of setting a `pod.Spec.TerminationGracePeriodSeconds` of 0 seconds is unsafe and strongly discouraged for Stateful Set Pods. Graceful deletion is safe and will ensure that the [Pod shuts down gracefully](/docs/user-guide/pods/#termination-of-pods) before the kubelet deletes the name from the apiserver. 
+
+Kubernetes (versions 1.5 or newer) will not delete Pods just because a Node is unreachable. The Pods running on an unreachable Node enter the 'Terminating' or 'Unknown' state eventually. Pods may also enter these states when the user attempts graceful deletion (*link to kubectl documentation*) of a pod on an unreachable Node. The only ways in which a pod in such a state can be removed from the apiserver are as follows:
```

Comment:
Do not refer to the reader as "user" in indirect third person. Address the reader directly as "you."

"Pods may also enter these states when you attempt to gracefully terminate a pod on an unreachable node."